### PR TITLE
Add "Delay" column to Query Log

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Discourse Feature request: Add "Delay" column to Query Log (as it was with v5)

Screenshot from 2023-10-16 16-48-01

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: scripts/pi-hole/js/queries.js [hunk 1]
```diff
@@ -208,6 +208,24 @@ function parseQueryStatus(data) {
...
+    //  - milliseconds if reply time >= 100 µs
+    //  - microseconds otherwise
+    return replyTime < 1e-4
+      ? (1e6 * replyTime).toFixed(1) + " µs"
+      : replyTime < 1
+      ? (1e3 * replyTime).toFixed(1) + " ms"
+      : replyTime.toFixed(1) + " s";
+  }
+
+  // else: return the number itself (for sorting and searching)
```
Comment (yubiuser) [inline scripts/pi-hole/js/queries.js:224]: Sorting does not work at all

Screenshot at 2023-10-16 21-59-26

File: scripts/pi-hole/js/queries.js [hunk 2]
```diff
@@ -306,11 +324,7 @@ function formatInfo(data) {
   if (data.reply.type !== "UNKNOWN") {
     replyInfo = divStart + "Reply:&nbsp&nbsp;" + data.reply.type;
     if (data.reply.time >= 0 && data.reply.type !== "UNKNOWN") {
-      replyInfo +=
-        " (" +
-        (data.reply.time < 1
-          ? (1e3 * data.reply.time).toFixed(1) + " ms)"
-          : data.reply.time.toFixed(1) + " s)");
+      replyInfo += " (" + formatReplyTime(data, "display") + ")";
```
Comment (yubiuser) [inline scripts/pi-hole/js/queries.js]: I don't think this info is necessary. It only doubles the info we have already in the delay column

**Review by yubiuser — COMMENTED** [review]
Maybe 'Delay' is not the best phrase to use. Users could think Pi-hole is delaying query resolution and will start to ask how to reduce delay. Maybe 'Reply time' or simply 'time' or 'query time' would be better

**Comment by DL6ER** [comment]
I have been thinking about the name, too.
- `Time` is already used on the very left - even if we'd change it is "timestamp", "time" still doesn't fit in here IMO
- any combination ("reply time", "query time", etc.) will break the line and make the header double in height. This looked very ugly in my tests.

**Comment by yubiuser** [comment]
> any combination ("reply time", "query time", etc.) will break the line and make the header double in height. This looked very ugly in my tests.

'Type' could give some space to be used in 'Query time'

**Comment by rdwebdesign** [comment]
Maybe `R. Time`? 

Or
```
<th title="Reply Time">Time</th>
```

**Comment by DL6ER** [comment]
Actually I don't know how to steal width from type. It is not intentionally that big.

@yubiuser Do you remember when you asked why I added all the additional sorting possibilities we aren't using right now in FTL? We just found **the one** that was missing: https://github.com/pi-hole/FTL/pull/1671

File: scripts/pi-hole/js/queries.js (hunk 1 above)
Comment (DL6ER) [reply scripts/pi-hole/js/queries.js:224]: It needs https://github.com/pi-hole/FTL/pull/1671

Comment (yubiuser) [reply scripts/pi-hole/js/queries.js:224]: Works with https://github.com/pi-hole/FTL/pull/1671

**Comment by rdwebdesign** [comment]
> I don't know how to steal width from type. It is not intentionally that big.

This is coming from here:
https://github.com/pi-hole/web/blob/3dd57522e2079622a19adda88a639a86485b4d3d/style/pi-hole.css#L88-L90

The third column was **_Domain_** column in v5, but the CSS was never updated. We can also fix/remove line 84.

**Comment by rdwebdesign** [comment]
About the "Delay" name: 
If you want to change the column name, but also use less space maybe we could use an icon:

image

**Comment by DL6ER** [comment]
Review comments addressed, I like the idea of going forward with an icon having a hover text for the curious and fixed the CSS to get rid of the wide type column.

**Review by rdwebdesign — APPROVED** [review]

**Review by yubiuser — APPROVED** [review]

**Pull request merged**